platforms/platform_db/platform_defaults.json

```diff
@@ -59,10 +48,106 @@
+                     "See num-timing-reg-stages above for a discussion.",
+                     "",
+                     "In addition to integer counts, the string 'auto' (use double",
+                     "quotes) will request (num-timing-reg-stages - 1) auto-inserted",
+                     "stages.  One is subtracted because the AFU is still expected",
+                     "to register the CCI-P signals.  Automatic mode shifts",
+                     "responsibility for managing variations in num-timing-reg-stages",
+                     "from AFUs to the platform."
+                  ],
+               "num-auto-timing-reg-stages": null,
+
+               "comment":
+                  [
+                     "AFUs may request that the platform transform the CCI-P clock",
+                     "to a different clock domain by setting this value to the",
+                     "name of another incoming clock."
+                  ],
+               "clock": "default"
+            },
+
+         "local-memory":
+            {
+               "comment":
+                  [
+                     "These properties all apply to a single bank.  It is assumed that",
+                     "all banks are configured identically."
+                  ],
+               "addr-width": 26,
+               "data-width": 512,
+               "burst-cnt-width": 7,
+
+               "comment":
+                  [
+                     "Like the CCI-P num-timing-reg-stages, this is the",
+                     "recommended number of times an AFU is expected to register",
+                     "local memory signals before use.  For Avalon MM interfaces",
+                     "this value is typically zero since registering Avalon MM's",
+                     "waitrequest protocol is complex.  The platform provides a",
+                     "module for adding pipeline stages when using the SystemVerilog",
+                     "Avalon interface:  avalon_mem_if_reg."
+                  ],
+               "num-timing-reg-stages": 0,
+
+               "comment":
+                  [
+                     "Like the CCI-P num-auto-timing-reg-stages, this field requests",
+                     "that the platform add pipeline stages to the local memory",
+                     "signals before passing them to the AFU.",
+                     "",
+                     "In addition to integer counts, the string 'auto' (use double",
+                     "quotes) will request num-timing-reg-stages auto-inserted",
+                     "stages.  Note the subtle difference from the CCI-P version.",
+                     "In CCI-P, one is subtracted in 'auto' mode because the AFU",
+                     "is expected to add a CCI-P register stage.  For local memory",
+                     "the AFU is not required to add a register stage, so the full",
+                     "count of num-timing-reg-stages is added by the platform."
```

Comment:
You appear to suggest that num-auto-timing-reg-stages specifies the number of pipeline stages added on the BBS side of the PR boundary. I dont understand how you can change BBS pipeline stages during AFU compile. Even if the AFU developer changes this value, it wont have any effect on the GBS. We dont let customers do BBS compiles.

Reply:
It's between ase_top (or green_bs) and the AFU, not in the BBS.

Reply:
ok. one more question. how is waitrequest handled if i set num-timing-reg-stages for memory to be non-zero value?

Reply:
It instantiates an Avalon-MM Pipeline Bridge that implements the protocol.